src/neo/Trie/MPT/MPTNode/ExtensionNode.cs

```diff
@@ -0,0 +1,41 @@
+using Neo.Cryptography;
+using Neo.IO;
+using Neo.IO.Json;
+using System.IO;
+
+namespace Neo.Trie.MPT
+{
+    public class ExtensionNode : MPTNode
+    {
+        public byte[] Key;
+        public MPTNode Next;
+
+        public ExtensionNode()
+        {
+            nType = NodeType.ExtensionNode;
+        }
+
+        public override void EncodeSpecific(BinaryWriter writer)
+        {
+            writer.WriteVarBytes(Key);
```

Comment:
We should define the max key length

Reply:
If we use the Constract Script Hash + Hash(StorageKey), all path will have fixed length and considering the `ToNibbles` the max key length `UInt256.Size * 4`.

Reply:
It should be contract id and storage keys. We don't use script hashes in states.